{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# RNA-seqの取り出し\n",
    "\n",
    " - 理研ID（Sxxxxxx）のリストを読み込む\n",
    " - STAR\n",
    "       対応するRNA-seq（Sxxxxx.ReadsPerGene.out.tab）のcountsを集計する\n",
    " - RSEM\n",
    "       Sxxxxx.genes.results, Sxxxxx.isoforms.resultsのexpected_countsを集計する\n",
    "\n",
    "[参考]\n",
    "- [STAR-RSEMによる発現量推定](https://rnakato.hatenablog.jp/entry/2018/12/28/184301)\n",
    "- [Quantification using RSEM](https://ycl6.gitbook.io/rna-seq-data-analysis/rna-seq_analysis_workflow/quantification_using_rsem1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# setup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "import glob\n",
    "import pandas as pd\n",
    "\n",
    "#RNA_seq_dir = '/interim/rnaseq-automated/GRCh38/'\n",
    "RNA_seq_path = './RNA-seq/'\n",
    "\n",
    "rikenID_file='prurigo_rikenID_bulk.csv'\n",
    "\n",
    "# 設定ファイルを読んでrikenIDリスト作成\n",
    "rikenids = pd.read_csv(rikenID_file,header=None)[0].to_list()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['S09796_L1', 'S09797', 'S09798']\n"
     ]
    }
   ],
   "source": [
    "# 全RNAseqを対象にする\n",
    "# 全ディレクトリ名を取得\n",
    "files = os.listdir(RNA_seq_path)\n",
    "#files_dir = [f for f in files if os.path.isdir(os.path.join(RNA_seq_path, f))]\n",
    "files_dir = [f for f in files if os.path.isdir(os.path.join(RNA_seq_path, f))]\n",
    "\n",
    "files_dir.sort()\n",
    "rikenids = files_dir\n",
    "pd.Series(rikenids).to_csv('RNAseq_sampleID_list.csv',index=False,header=False)\n",
    "\n",
    "print(files_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WASH7P</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIR6859-3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIR6859-2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIR6859-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           0  1  2  3\n",
       "0    DDX11L1  4  5  6\n",
       "1     WASH7P  1  0  1\n",
       "2  MIR6859-3  0  0  0\n",
       "3  MIR6859-2  0  0  0\n",
       "4  MIR6859-1  0  0  0"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# .ReadsPerGene.out.tab\n",
    "fileName = '.ReadsPerGene.out.tab'\n",
    "file_path = RNA_seq_path+'S09797'+'/'+'S09797'+fileName\n",
    "if os.path.exists(file_path):\n",
    "    df_tmp = pd.read_table(file_path,header=None,skiprows=4)\n",
    "df_tmp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "               0  1  2  3\n",
      "0        DDX11L1  4  5  6\n",
      "1         WASH7P  1  0  1\n",
      "2      MIR6859-3  0  0  0\n",
      "3      MIR6859-2  0  0  0\n",
      "4      MIR6859-1  0  0  0\n",
      "...          ... .. .. ..\n",
      "26470      CDY1B  0  0  0\n",
      "26471       CDY1  0  0  0\n",
      "26472   CSPG4P1Y  0  0  0\n",
      "26473  GOLGA2P3Y  0  0  0\n",
      "26474  GOLGA2P2Y  9  0  0\n",
      "\n",
      "[26475 rows x 4 columns]\n"
     ]
    }
   ],
   "source": [
    "print(df_tmp)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# STAR (ReadsPerGene.out.tab)\n",
    "ReadsPerGene.out.tabについて、geneIDを縦列、サンプル（理研ID）を横列としたRNA-seq countsのテーブルを作成する\n",
    "\n",
    " - ReadsPerGene.out.tab file with 4 columns which correspond to different strandedness options:\n",
    "1. column 1: gene ID\n",
    "2. column 2: counts for unstranded RNA-seq\n",
    "3. column 3: counts for the 1st read strand aligned with RNA (htseq-count option -s yes)\n",
    "4. column 4: counts for the 2nd read strand aligned with RNA (htseq-count option -s reverse)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "./RNA-seq/S09798/S09798.ReadsPerGene.out.tab : join ID =  ['S09796_L1', 'S09797', 'S09798']\n",
      "./RNA-seq/S09798/S09798.ReadsPerGene.out.tab : miss ID =  []\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S09796_L1</th>\n",
       "      <th>S09797</th>\n",
       "      <th>S09798</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>DDX11L1</th>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WASH7P</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MIR6859-3</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MIR6859-2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MIR6859-1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           S09796_L1  S09797  S09798\n",
       "gene_id                             \n",
       "DDX11L1            1       4       7\n",
       "WASH7P             0       1       1\n",
       "MIR6859-3          0       0       0\n",
       "MIR6859-2          0       0       0\n",
       "MIR6859-1          0       0       0"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# rikenIDがついたReadsPerGene.out.tabを見つけてread\n",
    "fileName = '.ReadsPerGene.out.tab'\n",
    "\n",
    "first_flag = True\n",
    "join_list = []\n",
    "miss_list = []\n",
    "\n",
    "\n",
    "\n",
    "for rikenid in rikenids:\n",
    "    \n",
    "    file_path = RNA_seq_path+rikenid+'/'+rikenid+fileName\n",
    "\n",
    "    if os.path.exists(file_path):\n",
    "        df_tmp = pd.read_table(file_path,header=None,skiprows=4)\n",
    "        # column=0 :gene ID\n",
    "        df_tmp = df_tmp.rename(columns={0:'gene_id',1:rikenid}).set_index('gene_id')\n",
    "        # column=1 :counts for unstranded RNA-seq\n",
    "        df_tmp = df_tmp[[rikenid]]\n",
    "        \n",
    "        # 初回のみ\n",
    "        if(first_flag):\n",
    "            df = df_tmp\n",
    "            first_flag = False\n",
    "            join_list.append(rikenid)\n",
    "        else:\n",
    "            # ２回目以降\n",
    "            df = df.join(df_tmp)\n",
    "            join_list.append(rikenid)\n",
    "    else:\n",
    "        miss_list.append(rikenid)\n",
    "\n",
    "print(file_path+' : join ID = ',join_list)\n",
    "print(file_path+' : miss ID = ',miss_list)\n",
    "\n",
    "df.to_csv('ReadsPerGene.out.tab_RNA-seq_counts.csv')\n",
    "df.head()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# RSEM (.genes.results, .isoforms.results)\n",
    "geneIDをexpected_counts縦列、サンプル（理研ID）を横列のテーブルを作成する\n",
    "\n",
    "- .genes.results\n",
    "1. gene_id\n",
    "2. transcript_id(s)\n",
    "3. length\n",
    "4. effective_length\n",
    "5. expected_count\n",
    "6. TPM\n",
    "7. FPKM\n",
    "\n",
    "- .isoforms.results\n",
    "1. transcript_id\n",
    "2. gene_id\n",
    "3. length\n",
    "4. effective_length\n",
    "5. expected_count\n",
    "6. TPM\n",
    "7. FPKM\n",
    "8. IsoPct"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S09798</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_id</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>DDX11L1</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WASH7P</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MIR6859-3</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MIR6859-2</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MIR6859-1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           S09798\n",
       "gene_id          \n",
       "DDX11L1         7\n",
       "WASH7P          1\n",
       "MIR6859-3       0\n",
       "MIR6859-2       0\n",
       "MIR6859-1       0"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# どんな内容か？\n",
    "fileName = '.genes.results'\n",
    "file_path = RNA_seq_path+'S09796'+'/'+'S09796'+fileName\n",
    "if os.path.exists(file_path):\n",
    "    df_tmp = pd.read_table(file_path,)\n",
    "df_tmp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>length</th>\n",
       "      <th>effective_length</th>\n",
       "      <th>expected_count</th>\n",
       "      <th>TPM</th>\n",
       "      <th>FPKM</th>\n",
       "      <th>IsoPct</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000373020</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>2206</td>\n",
       "      <td>100.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000494424</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>820</td>\n",
       "      <td>611.20</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000496771</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>1025</td>\n",
       "      <td>816.20</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000612152</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>3796</td>\n",
       "      <td>3587.20</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000614008</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>900</td>\n",
       "      <td>691.20</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENST00000373031</td>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>1339</td>\n",
       "      <td>1130.20</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENST00000485971</td>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>542</td>\n",
       "      <td>333.26</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ENST00000371582</td>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>1161</td>\n",
       "      <td>952.20</td>\n",
       "      <td>16.56</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ENST00000371584</td>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>1073</td>\n",
       "      <td>864.20</td>\n",
       "      <td>8.54</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.41</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     transcript_id          gene_id  length  effective_length  expected_count  \\\n",
       "0  ENST00000373020  ENSG00000000003    2206            100.00            1.00   \n",
       "1  ENST00000494424  ENSG00000000003     820            611.20            0.00   \n",
       "2  ENST00000496771  ENSG00000000003    1025            816.20            0.00   \n",
       "3  ENST00000612152  ENSG00000000003    3796           3587.20            0.00   \n",
       "4  ENST00000614008  ENSG00000000003     900            691.20            0.00   \n",
       "5  ENST00000373031  ENSG00000000005    1339           1130.20            0.00   \n",
       "6  ENST00000485971  ENSG00000000005     542            333.26            0.00   \n",
       "7  ENST00000371582  ENSG00000000419    1161            952.20           16.56   \n",
       "8  ENST00000371584  ENSG00000000419    1073            864.20            8.54   \n",
       "\n",
       "    TPM  FPKM  IsoPct  \n",
       "0  0.00  0.01  100.00  \n",
       "1  0.00  0.00    0.00  \n",
       "2  0.00  0.00    0.00  \n",
       "3  0.00  0.00    0.00  \n",
       "4  0.00  0.00    0.00  \n",
       "5  0.00  0.00    0.00  \n",
       "6  0.00  0.00    0.00  \n",
       "7  0.10  0.27    0.72  \n",
       "8  0.06  0.15    0.41  "
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# どんな内容か？\n",
    "fileName = '.isoforms.results'\n",
    "file_path = RNA_seq_path+'S09796'+'/'+'S09796'+fileName\n",
    "if os.path.exists(file_path):\n",
    "    df_tmp = pd.read_table(file_path)\n",
    "df_tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ".genes.results : join ID =  ['S09796_L1', 'S09798', 'S09797']\n",
      ".genes.results : miss ID =  []\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S09796_L1</th>\n",
       "      <th>S09798</th>\n",
       "      <th>S09797</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000000003</th>\n",
       "      <td>40.20</td>\n",
       "      <td>40.20</td>\n",
       "      <td>40.20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000005</th>\n",
       "      <td>1.21</td>\n",
       "      <td>1.21</td>\n",
       "      <td>1.21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000419</th>\n",
       "      <td>32.34</td>\n",
       "      <td>32.34</td>\n",
       "      <td>32.34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000457</th>\n",
       "      <td>2.32</td>\n",
       "      <td>2.32</td>\n",
       "      <td>2.32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000460</th>\n",
       "      <td>18.38</td>\n",
       "      <td>18.38</td>\n",
       "      <td>18.38</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 S09796_L1  S09798  S09797\n",
       "gene_id                                   \n",
       "ENSG00000000003      40.20   40.20   40.20\n",
       "ENSG00000000005       1.21    1.21    1.21\n",
       "ENSG00000000419      32.34   32.34   32.34\n",
       "ENSG00000000457       2.32    2.32    2.32\n",
       "ENSG00000000460      18.38   18.38   18.38"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# rikenIDがついた.genes.resultsを見つけてread\n",
    "fileName = '.genes.results'\n",
    "target_column = 'TPM'\n",
    "join_list = []\n",
    "miss_list = []\n",
    "first_flag = True\n",
    "\n",
    "for rikenid in rikenids:\n",
    "\n",
    "    file_path = RNA_seq_path+rikenid+'/'+rikenid+fileName\n",
    "\n",
    "    if os.path.exists(file_path):\n",
    "        df_tmp = pd.read_table(file_path)\n",
    "        # column=0 :gene_id\n",
    "        df_tmp = df_tmp.rename(columns={target_column:rikenid}).set_index('gene_id')\n",
    "        # column=1 :counts for unstranded RNA-seq\n",
    "        df_tmp = df_tmp[[rikenid]]\n",
    "\n",
    "        # 初回のみ\n",
    "        if(first_flag):\n",
    "            df = df_tmp\n",
    "            first_flag = False\n",
    "            join_list.append(rikenid)\n",
    "        else:\n",
    "            # ２回目以降\n",
    "            df = df.join(df_tmp)\n",
    "            join_list.append(rikenid)\n",
    "    else:\n",
    "        miss_list.append(rikenid)\n",
    "\n",
    "print(fileName+' : join ID = ',join_list)\n",
    "print(fileName+' : miss ID = ',miss_list)\n",
    "\n",
    "df.to_csv('genes.results_expected_count.csv')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ".isoforms.results : join ID =  ['S09796', 'S09797', 'S09798']\n",
      ".isoforms.results : miss ID =  ['S10869', 'S10870', 'S15431']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>S09796</th>\n",
       "      <th>S09797</th>\n",
       "      <th>S09798</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENST00000373020</th>\n",
       "      <th>ENSG00000000003</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENST00000494424</th>\n",
       "      <th>ENSG00000000003</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENST00000496771</th>\n",
       "      <th>ENSG00000000003</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENST00000612152</th>\n",
       "      <th>ENSG00000000003</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENST00000614008</th>\n",
       "      <th>ENSG00000000003</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 S09796  S09797  S09798\n",
       "transcript_id   gene_id                                \n",
       "ENST00000373020 ENSG00000000003     1.0     1.0     1.0\n",
       "ENST00000494424 ENSG00000000003     0.0     0.0     0.0\n",
       "ENST00000496771 ENSG00000000003     0.0     0.0     0.0\n",
       "ENST00000612152 ENSG00000000003     0.0     0.0     0.0\n",
       "ENST00000614008 ENSG00000000003     0.0     0.0     0.0"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fileName = '.isoforms.results'\n",
    "join_list = []\n",
    "miss_list = []\n",
    "first_flag = True\n",
    "\n",
    "for rikenid in rikenids:\n",
    "\n",
    "    file_path = RNA_seq_path+rikenid+'/'+rikenid+fileName\n",
    "\n",
    "    if os.path.exists(file_path):\n",
    "        df_tmp = pd.read_table(file_path)\n",
    "        # column=0 :gene_id\n",
    "        df_tmp = df_tmp.rename(columns={'expected_count':rikenid}).set_index(['transcript_id','gene_id'])\n",
    "        # column=1 :counts for unstranded RNA-seq\n",
    "        df_tmp = df_tmp[[rikenid]]\n",
    "\n",
    "        # 初回のみ\n",
    "        if(first_flag):\n",
    "            df = df_tmp\n",
    "            first_flag = False\n",
    "            join_list.append(rikenid)\n",
    "        else:\n",
    "            # ２回目以降\n",
    "            df = df.join(df_tmp)\n",
    "            join_list.append(rikenid)\n",
    "    else:\n",
    "        miss_list.append(rikenid)\n",
    "\n",
    "print(fileName+' : join ID = ',join_list)\n",
    "print(fileName+' : miss ID = ',miss_list)\n",
    "\n",
    "df.to_csv('isoforms.results_expected_count.csv')\n",
    "df.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
